### ISSUE TITLE
[Request]: add assethash fix like stand

### ISSUE THREAD

**Issue by MsLGXC** [Yimura/YimMenu#2442]
### Solution

Im a mod creater for GTA，For many years, I have been committed to modify mod in story mode to an online version available for everyone to use.In the past two years, I have achieved great results. Not only  I can use mods online,even shv asi, shvdn .net mod without any menu.but I also disguise all assets as the original dlc of the game by modify the dlc setup file, which reduces the space occupied by mods to almost the same size as the source file.

Unfortunately, although Rock Star has almost acquiesced in your use of mod. they still added asset hash checks to the game. For all DLCs except for titleupdate, rockstar has added hashcheck. Surprisingly, you can use the replacement method to replace them. Hashcheck only checks for newly added items and does not check for existing files one by one, even if they are clearly different. In addition to adding new files, hashcheck also checks most meta type files  and replace will be detection(meta check also inc updaterpf).Especially things that directly affect gameplay, such as weapon parameters, vehicle parameters, NPC parameters, etc

In this case, you can add up to approximately 6-7g of additional files (as a single RPF can only read up to 4GB, otherwise the game engine will make an error) and cannot modify any meta type files
If you choose to modify the added mod file to another DLC except titleupdate, the game will verify that your assets are different from other people, but you can still play, but you cannot join any session (unless that person and you have installed the same thing)

I searched for  4month for assethashcheck's solution , but it was in vain. My final conclusion is that I cannot simply remove verification through modify certain files without using the menu hook game.But I have no knowledge about making cheat or programs, I am just a poor mod producer.So I hope to seek help from the open source community.

### Reason

Indeed, I can simply use Stand to directly solve this problem without paying attention to others, and even bundle and sell Stand to them. I have always been a loyal user of Stand, writing some shit lua and help some people translate theirs lua

But is this really what I want?
I don't want to tell everyone that if you want to enjoy graphmod in online, add on maps, or even crazier offline modules. You must pay me and purchase a stand, otherwise you cannot play with other people.

I hope everyone can enjoy the benefits of this feature,just like Yim, which is the meaning and spirit of the open source community. 

I understand that this feature may be difficult to implement, otherwise it should quickly appear on all paid menus and then on the free menu. But I still hold hope and request Yim to make such an attempt.

**Comment by umorabi** [comment]
+1

**Comment by Unnkai163** [comment]
+1

**Comment by maybegreat48** [comment]
Fine, I'll try to add that if you're ready to test it. How would you change your asset hash? Do you load a DLC with OpenIV?

**Comment by Shendu9** [comment]
Very good project, love from China

**Comment by MsLGXC** [comment]
> Fine, I'll try to add that if you're ready to test it. How would you change your asset hash? Do you load a DLC with OpenIV?

I have sufficient time to test, and I can provide a stand account for you to use, but I cannot provide any code advice. After all, I am a 'modwalker' not a 'codewalker'. I will describe as much as possible about all the information I know

**Comment by MsLGXC** [comment]
> Fine, I'll try to add that if you're ready to test it. How would you change your asset hash? Do you load a DLC with OpenIV?

To be precise, I do not directly add a DLC to enter the game
This is because if you have additional DLC, the game will notify you that an additional update has been installed and force you to return to story mode, preventing you from entering online mode

**Comment by maybegreat48** [comment]
> > Fine, I'll try to add that if you're ready to test it. How would you change your asset hash? Do you load a DLC with OpenIV?
> 
> To be precise, I do not directly add a DLC to enter the game This is because if you have additional DLC, the game will notify you that an additional update has been installed and force you to return to story mode, preventing you from entering online mode

Can you show the exact alert text?

**Comment by MsLGXC** [comment]
> 好吧，如果你准备好测试它，我会尝试添加它。您将如何更改资产哈希值？您是否使用 OpenIV 加载 DLC？

Therefore, without any menu, I adopted another approach. For the above, I have extracted all the files for the added DLC and manually written them into the installation file in update.rpf. The game does not verify the titleupdate dlc, perhaps because it is a hot update dlc. Then the game will read these newly added files in the normal way and due to lack of verification, the game will consider them as the files added by the official update

**Comment by MsLGXC** [comment]
> Fine, I'll try to add that if you're ready to test it. How would you change your asset hash? Do you load a DLC with OpenIV?

Therefore, without any menu, I adopted another approach. For the above, I have extracted all the files for the added DLC and manually written them into the installation file in update.rpf. The game does not verify the titleupdate dlc, perhaps because it is a hot update dlc. Then the game will read these newly added files in the normal way and due to lack of verification, the game will consider them as the files added by the official update

**Comment by MsLGXC** [comment]
> > > Fine, I'll try to add that if you're ready to test it. How would you change your asset hash? Do you load a DLC with OpenIV?
> > 
> > 
> > To be precise, I do not directly add a DLC to enter the game This is because if you have additional DLC, the game will notify you that an additional update has been installed and force you to return to story mode, preventing you from entering online mode
> 
> Can you show the exact alert text?

of course wait some time

**Comment by tupoy-ya** [comment]
Seems related:
```cpp
// pointers.cpp
// Not in YimMenu format.
main_batch.add("Get Data Hash", "48 89 ? ? ? 55 56 57 48 83 ? ? 33 ED 8B", [this](memory::handle ptr) {
	m_get_data_hash = ptr.as<PVOID>();
});

// hooks/misc/get_data_hash.cpp
#include "hooking.hpp"

#include <network/Network.hpp>

namespace big
{
	uint32_t hooks::get_data_hash(uint64_t p0, uint32_t p1)
	{
		uint32_t result = g_hooking->get_original<get_data_hash>()(p0, p1);
		LOG(G3LOG_DEBUG) << result << " = hooks::get_data_hash( " << p0 << ", " << p1 << ");";
		return g.settings.data_hash;
	}
}
```


The return value changes when i load the game with rpf mods.
I get `4018721125` when i load without any extra rpfs.
Changing the return value makes you load into solo session, however setting to `4018721125` does not allow you to join public sessions with rps mods.

The function itself seems to be iterating over the dlcs.

### EDIT:
Originally written for #1478 some things could be out of date.

As for the warning message, i remove it by forcing `GET_EVER_HAD_BAD_PACK_ORDER` to return false.
Which is a bad way to go around this, but maybe it will help someone.

**Comment by MsLGXC** [comment]
However, I don't think DLC is the key because I can "graft" it onto the original DLC using the above method. Even if you prevent the game from verifying DLC, the game will still verify assethash, unless you can prevent the game from doing both things at the same time, which would be great

**Comment by Tessio** [comment]
> Fine, I'll try to add that if you're ready to test it. How would you change your asset hash? Do you load a DLC with OpenIV?

is what I was referring to in the #2343 
Count on me for testing or more information

![image](https://github.com/YimMenu/YimMenu/assets/50866332/2190269a-975c-4bdb-874f-2ea32d28ff79)

**Comment by Tessio** [comment]
By adding these, it would give us the opportunity to add entire addon dlcs, in addition to being able to modify YMT, Meta, YTF, etc.

**Comment by MsLGXC** [comment]
> By adding these, it would give us the opportunity to add entire addon dlcs, in addition to being able to modify YMT, Meta, YTF, etc.

Yes, I have already done so. Everything offline can be brought online. Stand still cannot load added maps or modify the entire world. but I have already done so well for this all, the only drawback is that you must use stand fixhash, otherwise you can only play with people with the same mod.
This is exactly what I have requested, allowing everyone to enjoy it without the help of a paid menu

**Comment by MsLGXC** [comment]
And I remove the online verification of shv based on the ancient tutorials of uc forum, which allows everyone to use. asi and. net plugins online. R Star does not verify the predecessor of dinput, dsound, so you can easily load any plugin and have dsound use the mods folder of OIV for gaming. Just like sp.

**Comment by MsLGXC** [comment]
> And I remove the online verification of shv based on the ancient tutorials of uc forum, which allows everyone to use. asi and. net plugins online. R Star does not verify the predecessor of dinput, dsound, so you can easily load any plugin and have dsound use the mods folder of OIV for gaming. Just like sp.

It's interesting that using various SP Trainer and plugins is even safer than using menus, and R Star doesn't care about these at all

**Comment by MsLGXC** [comment]
> > > Fine, I'll try to add that if you're ready to test it. How would you change your asset hash? Do you load a DLC with OpenIV?
> > 
> > 
> > To be precise, I do not directly add a DLC to enter the game This is because if you have additional DLC, the game will notify you that an additional update has been installed and force you to return to story mode, preventing you from entering online mode
> 
> Can you show the exact alert text?

I don't know why, the old prompt seems to have been replaced with a time out, and I am trying to find the original prompt

**Comment by Tessio** [comment]
> > > > Fine, I'll try to add that if you're ready to test it. How would you change your asset hash? Do you load a DLC with OpenIV?
> > > 
> > > 
> > > To be precise, I do not directly add a DLC to enter the game This is because if you have additional DLC, the game will notify you that an additional update has been installed and force you to return to story mode, preventing you from entering online mode
> > 
> > 
> > Can you show the exact alert text?
> 
> I don't know why, the old prompt seems to have been replaced with a time out, and I am trying to find the original prompt

I remember one of the error messages was something to the effect that there is a new update or something like that.

**Comment by MsLGXC** [comment]
> > > > > Fine, I'll try to add that if you're ready to test it. How would you change your asset hash? Do you load a DLC with OpenIV?
> > > > 
> > > > 
> > > > To be precise, I do not directly add a DLC to enter the game This is because if you have additional DLC, the game will notify you that an additional update has been installed and force you to return to story mode, preventing you from entering online mode
> > > 
> > > 
> > > Can you show the exact alert text?
> > 
> > 
> > I don't know why, the old prompt seems to have been replaced with a time out, and I am trying to find the original prompt
> 
> I remember one of the error messages was something to the effect that there is a new update or something like that.

yeah that  is addondlc's alert,im looking for diffasset's alert

**Comment by MsLGXC** [comment]
![1 (2)](https://github.com/YimMenu/YimMenu/assets/94236627/681d8c51-4dc3-4407-a5f8-44459f174cb3)
this is addondlc alert
still looking for assetalert

**Comment by maybegreat48** [comment]
Can anyone here test https://github.com/YimMenu/YimMenu/pull/2443?

**Comment by Killer4563782** [comment]
> Fine, I'll try to add that if you're ready to test it. How would you change your asset hash? Do you load a DLC with OpenIV?

Wouldn't it be beneficial to add Script Hook V? It gives you the ability to load offline mods and edit textures. This would enable you to load custom cars or addon cars.

**Comment by MsLGXC** [comment]
> Can anyone here test #2443?

ok im doing other maybe need half day and i think asset alert has been replace a new tip
![1111](https://github.com/YimMenu/YimMenu/assets/94236627/b7521e40-3f06-4c11-b2c0-8d48b25ff9c6)

**Comment by Tessio** [comment]
I obtained the same result

> > Can anyone here test #2443?
> 
> ok im doing other maybe need half day and i think asset alert has been replace a new tip ![1111](https://user-images.githubusercontent.com/94236627/284923223-b7521e40-3f06-4c11-b2c0-8d48b25ff9c6.png)

**Comment by Tessio** [comment]
> > Fine, I'll try to add that if you're ready to test it. How would you change your asset hash? Do you load a DLC with OpenIV?
> 
> Wouldn't it be beneficial to add Script Hook V? It gives you the ability to load offline mods and edit textures. This would enable you to load custom cars or addon cars.

With the asset, we would not need scripthook to load addons.

**Comment by MsLGXC** [comment]
> > Fine, I'll try to add that if you're ready to test it. How would you change your asset hash? Do you load a DLC with OpenIV?
> 
> Wouldn't it be beneficial to add Script Hook V? It gives you the ability to load offline mods and edit textures. This would enable you to load custom cars or addon cars.

No, that's not the use of shv，load mod or edit textures no need shv.Plugins that only call game internal functions require it, such as trainers.You can also use a completely local method to add additional vehicles without using SHVs, but that can be complex, so usually we use SHVs to allow trainers to directly recognize these added items and generate them

**Comment by MsLGXC** [comment]
> > > Fine, I'll try to add that if you're ready to test it. How would you change your asset hash? Do you load a DLC with OpenIV?
> > 
> > 
> > Wouldn't it be beneficial to add Script Hook V? It gives you the ability to load offline mods and edit textures. This would enable you to load custom cars or addon cars.
> 
> With the asset, we would not need scripthook to load addons.

No, the plugin still requires shv support. But asset repair will allow you to modify and add anything that R Star does not want you to modify, as you mentioned above

**Comment by MsLGXC** [comment]
> By adding these, it would give us the opportunity to add entire addon dlcs, in addition to being able to modify YMT, Meta, YTF, etc.

just like u say,i can do this all the time but if havent hashfix,we can only play solo cant join anyone.hashfix will not affect the importance of shv.Without shv, all of your ASIs, even your ENBs, would be unusable.

**Comment by Tessio** [comment]
> > By adding these, it would give us the opportunity to add entire addon dlcs, in addition to being able to modify YMT, Meta, YTF, etc.
> 
> just like u say,i can do this all the time but if havent hashfix,we can only play solo cant join anyone.hashfix will not affect the importance of shv.Without shv, all of your ASIs, even your ENBs, would be unusable.

That is correct, hence the need to modify the assethash.

**Comment by maybegreat48** [comment]
Can you test now?

**Comment by MsLGXC** [comment]
> Can you test now?

im eating could u bulid dll file to me?

**Comment by gir489returns** [comment]
> > Can you test now?
> 
> im eating could u bulid dll file to me?

The build is still linked in the PR, when he makes subsequent commits, the build bot updates the comment it made originally with a new DLL.

**Comment by Tessio** [comment]
> Can you test now?

I'm working on it

**Comment by Tessio** [comment]
I tried installing an oiv of Euphoria Ragdoll, and it loaded perfectly the session with players, no error when joining, I confirm that it works, now I will try with Natural Vision Evolved, give me a moment.

**Comment by Unnkai163** [comment]
NVE graphics?
He probably is. An error occurred during join online. perhaps. Will crash your game.
You can try the add-on DLC cars. (This is the easiest way.)
Go to a public meeting if you can. Can see other players.
Then it was successful.

**Comment by Tessio** [comment]
> NVE graphics? He probably is. An error occurred during join online. perhaps. Will crash your game. You can try the add-on DLC cars. (This is the easiest way.) Go to a public meeting if you can. Can see other players. Then it was successful.

It should be possible, since it is possible with a stand

**Comment by MsLGXC** [comment]
im testing

**Comment by Tessio** [comment]
It works but as a recommendation for this function it is necessary to implement an Early Inject Enhancements function to add Game Pools Size Multiplier to 7.00 and Memory Pool Size MB 2147.
to avoid maintransition errors due to lack of memory with big addons. 

![Yim 2](https://github.com/YimMenu/YimMenu/assets/50866332/75384814-f20a-42a6-8fdb-af3f6fb2bbcf)
![YIM1](https://github.com/YimMenu/YimMenu/assets/50866332/626ee957-c063-422a-9467-80c89adfd0c3)
![YIM4](https://github.com/YimMenu/YimMenu/assets/50866332/1a96e124-51c2-4aff-a1d9-ace48f7ef291)

I had to use this Stand function to give more memory to the game, but the Assethash fix from YimMenu worked perfectly.

![yim3](https://github.com/YimMenu/YimMenu/assets/50866332/80e42811-78ee-4343-bc0f-51e40c6781ac)

**Comment by MsLGXC** [comment]
> > NVE graphics? He probably is. An error occurred during join online. perhaps. Will crash your game. You can try the add-on DLC cars. (This is the easiest way.) Go to a public meeting if you can. Can see other players. Then it was successful.
> 
> It should be possible, since it is possible with a stand

The ez way-put this flie in

> It works but as a recommendation for this function it is necessary to implement an Early Inject Enhancements function to add GAME POOLS SIZE Multiplier to 7.00 and Memory Pool Size MB 2147. to avoid maintransition errors due to lack of memory with big addons.
> 
> ![Yim 2](https://user-images.githubusercontent.com/50866332/285000722-75384814-f20a-42a6-8fdb-af3f6fb2bbcf.png) ![YIM1](https://user-images.githubusercontent.com/50866332/285000731-626ee957-c063-422a-9467-80c89adfd0c3.png) ![YIM4](https://user-images.githubusercontent.com/50866332/285000734-1a96e124-51c2-4aff-a1d9-ace48f7ef291.png)
> 
> I had to use this Stand function to give more memory to the game, but the Assethash fix from YimMenu worked perfectly.
> 
> ![yim3](https://user-images.githubusercontent.com/50866332/285001202-80e42811-78ee-4343-bc0f-51e40c6781ac.png)

u install nveo for zzoko or nve for razed ?

**Comment by Tessio** [comment]
> u install nveo for zzoko or nve for razed ?

Newest version October 31th Single Player

**Comment by Tessio** [comment]
> u install nveo for zzoko or nve for razed ?

What file?

**Comment by MsLGXC** [comment]
> > u install nveo for zzoko or nve for razed ?
> 
> What file?

Ignore it, this is a delay message

**Comment by Tessio** [comment]
We only need to implement the Game Pools Size and Memory Pool Size and we would have made history today my dear friends.

**Comment by MsLGXC** [comment]
[meta.zip](https://github.com/YimMenu/YimMenu/files/13443187/meta.zip)


> We only need to implement the Game Pools Size and Memory Pool Size and we would have made history today my dear friends.

uninstall other mod,then unzip this file to Grand Theft Auto V\x64\data,then try again

**Comment by MsLGXC** [comment]
> [meta.zip](https://github.com/YimMenu/YimMenu/files/13443187/meta.zip)
> 
> > We only need to implement the Game Pools Size and Memory Pool Size and we would have made history today my dear friends.
> 
> uninstall other mod,then unzip this file to Grand Theft Auto V\x64\data,then try again


> We only need to implement the Game Pools Size and Memory Pool Size and we would have made history today my dear friends.

Fine, I confirm that he has succeeded

**Comment by MsLGXC** [comment]
> We only need to implement the Game Pools Size and Memory Pool Size and we would have made history today my dear friends.

Actually, we have ready-made sp asi and do not need to write this function separately

**Comment by MsLGXC** [comment]
I will continue testing, and currently it seems to be working very well. I will try the addon map I have already made

**Comment by Tessio** [comment]
> > We only need to implement the Game Pools Size and Memory Pool Size and we would have made history today my dear friends.
> 
> Actually, we have ready-made sp asi and do not need to write this function separately

Which ".asi" meets these requirements?

**Comment by MsLGXC** [comment]
![`)D@02KLE89I$0NE5NT %K5](https://github.com/YimMenu/YimMenu/assets/94236627/0c15b8c6-368d-4d0e-84dd-98ab6a6dca76)
![JD%FKLTSM%AL0)GB~G%_XR](https://github.com/YimMenu/YimMenu/assets/94236627/23d18645-e38c-453f-8efb-50d49e90ea13)
I have many ASIs, they have many uses
But gameconfig is an important part of it
You only need to edit the file correctly

**Comment by MsLGXC** [comment]
my god

**Comment by MsLGXC** [comment]
![EA~ T46T1%T0WH{S%`T%T@X](https://github.com/YimMenu/YimMenu/assets/94236627/68d3aa64-b9e5-49a9-8916-59dded67eb2c)
![_1E{41``K)R9I7}JYA5I%TF](https://github.com/YimMenu/YimMenu/assets/94236627/935df903-8e76-4230-b643-aec9be181b5f)
![DEFF`UGDO6YUH26`SEB8PIH](https://github.com/YimMenu/YimMenu/assets/94236627/662b7508-4a26-461b-b1d1-1b5e92864a73)

**Comment by MsLGXC** [comment]
> ![44](https://user-images.githubusercontent.com/94236627/285012511-0db58152-4551-4b66-8998-9179e8f8320b.png) ![22](https://user-images.githubusercontent.com/94236627/285012553-c7b80152-aab8-48c8-91bf-fb39fc6f3910.png) ![33](https://user-images.githubusercontent.com/94236627/285012563-dcd665a6-81ed-4d41-a16d-f39a8f4096b0.png)

You succeeded! In such a short time!

**Comment by MsLGXC** [comment]
I ported a large number of maps and mods online six months ago, but there was no way for everyone to play for free except for Stand. Therefore, I gave up publishing and spent several months searching for a solution

**Comment by MsLGXC** [comment]
Congratulations,and thanksu!

**Comment by maybegreat48** [comment]
> I had to use this Stand function to give more memory to the game

Have you tried [this](https://github.com/maybegreat48/YimASI/tree/master/src)?

**Comment by Tessio** [comment]
> > I had to use this Stand function to give more memory to the game
> 
> Have you tried [this](https://github.com/maybegreat48/YimASI/tree/master/src)?

I am using an asi loader bypass in the gta v folder and HeapAdjuster.asi, PackfileLimitAdjuster It is still necessary to make Memory Pool for me, in heavy mods.

**Comment by maybegreat48** [comment]
Do you know the exact pools that have to be raised for your mod to work? I could add that to YimASI along with this thing

**Comment by pierrelasse** [comment]
Damn, i more comment and this is the top commented issue. Great work everyone :^)

**Comment by Tessio** [comment]
> > > I had to use this Stand function to give more memory to the game
> > 
> > 
> > Have you tried [this](https://github.com/maybegreat48/YimASI/tree/master/src)?
> 
> I am using an asi loader bypass in the gta v folder and HeapAdjuster.asi, PackfileLimitAdjuster It is still necessary to make Memory Pool for me, in heavy mods.

@MsLGXC is the one who could help with it

**Comment by Tessio** [comment]
> Do you know the exact pools that have to be raised for your mod to work? I could add that to YimASI along with this thing

@MsLGXC is the one who could help with it

**Comment by Tessio** [comment]
This is the error I get when I don't use stand to manage the pools and I try to enter online
[cout.log](https://github.com/YimMenu/YimMenu/files/13444207/cout.log)

**Comment by MsLGXC** [comment]
> > > I had to use this Stand function to give more memory to the game
> > 
> > 
> > Have you tried [this](https://github.com/maybegreat48/YimASI/tree/master/src)?
> 
> I am using an asi loader bypass in the gta v folder and HeapAdjuster.asi, PackfileLimitAdjuster It is still necessary to make Memory Pool for me, in heavy mods.

You also need res adj to increase the graphics memory allocation of the texture, otherwise the game will become a potato when you use high-resolution mapping mods. In addition, the correct gameconfig is required, and you can directly use this version or compare it with the game (if you want to integrate it into yim)

**Comment by MsLGXC** [comment]
[gameconfig.zip](https://github.com/YimMenu/YimMenu/files/13445700/gameconfig.zip)

**Comment by MsLGXC** [comment]
and use poolmanagerasi u can find what value crash u game

**Comment by tupoy-ya** [comment]
> Can anyone here test #2443?

It works, i can join a public session with mods, the main issue now is actually loading them because if you load [ClosedIV](https://github.com/martonp96/ClosedIV/) and YimMenu at the same time the game will crash.

To get it to load, you need to inject ClosedIV with dll hijacking(ASI Loader) and then inject YimMenu with an injector manually.
Maybe we can add rpf loading in a later PR.

**Comment by Tessio** [comment]
> [gameconfig.zip](https://github.com/YimMenu/YimMenu/files/13445700/gameconfig.zip)

Use your gameconfig.xml and poolmanager.asi to look for the possible crashes error.

You still can't load online without the 2 Stand functions activated.

Poolmanager.asi, no warning

**Comment by MsLGXC** [comment]
> > [gameconfig.zip](https://github.com/YimMenu/YimMenu/files/13445700/gameconfig.zip)
> 
> Use your gameconfig.xml and poolmanager.asi to look for the possible crashes error.
> 
> You still can't load online without the 2 Stand functions activated.
> 
> Poolmanager.asi, no warning

crash or what?

**Comment by Tessio** [comment]
> > > [gameconfig.zip](https://github.com/YimMenu/YimMenu/files/13445700/gameconfig.zip)
> > 
> > 
> > Use your gameconfig.xml and poolmanager.asi to look for the possible crashes error.
> > You still can't load online without the 2 Stand functions activated.
> > Poolmanager.asi, no warning
> 
> crash or what?

Crash

**Comment by MsLGXC** [comment]
> > > > [gameconfig.zip](https://github.com/YimMenu/YimMenu/files/13445700/gameconfig.zip)
> > > 
> > > 
> > > Use your gameconfig.xml and poolmanager.asi to look for the possible crashes error.
> > > You still can't load online without the 2 Stand functions activated.
> > > Poolmanager.asi, no warning
> > 
> > 
> > crash or what?
> 
> Crash

At least I have no problem using it, maybe due to the SP version, I am trying to create an online libertycity. After I finish, I will download newest and take a look at SP NVE again

**Comment by maybegreat48** [comment]
> You still can't load online without the 2 Stand functions activated.

Can you try enabling just one instead of both?

**Comment by MsLGXC** [comment]
![image](https://github.com/YimMenu/YimMenu/assets/94236627/2f6e1968-6efb-4fcb-be20-e01935b5dee3)
This is the mod I am currently using, with a thank you list on the left and all mods or plugins I have used or provided assistance on the right, but written in Chinese

**Comment by MsLGXC** [comment]
fully Realism Beyond
fully sg project
fully nbvisual
fully vremaster
fully Forests of San Andreas
and old modify ver nve maybe last year ver i forgot it
The above are larger and the original author used them as mods for adding DLC
There are many other mods that I won't explain one by one because they can be used online without any modifications

**Comment by MsLGXC** [comment]
> fully Realism Beyond fully sg project fully nbvisual fully vremaster fully Forests of San Andreas and old modify ver nve maybe last year ver i forgot it The above are larger and the original author used them as mods for adding DLC There are many other mods that I won't explain one by one because they can be used online without any modifications

There are also 5 additional maps available, including four from the ON3FLY3R Japan series
There is also a map is nordschleife

**Comment by MsLGXC** [comment]
I use all of these at the same time, about 100 mods, and the original file of mods alone is as high as 23g, not after installation
For me, I haven't encountered any problems yet

**Comment by maybegreat48** [comment]
> This is the mod I am currently using, with a thank you list on the left and all mods or plugins I have used or provided assistance on the right, but written in Chinese

I was talking about the Stand options that you mentioned earlier, the game and memory pools. Can you try enabling just one of those and see if you can go online? Can you also try to load a solo session instead of a public session?

> I use all of these at the same time, about 100 mods, and the original file of mods alone is as high as 23g, not after installation
For me, I haven't encountered any problems so far yet

Well I guess that explains it

**Comment by Tessio** [comment]
> I was talking about the Stand options that you mentioned earlier, the game and memory pools. Can you try enabling just one of those and see if you can go online? Can you also try to load a solo session instead of a public session?

If necessary I will record a video, now I will do the tests.

**Comment by MsLGXC** [comment]
> > This is the mod I am currently using, with a thank you list on the left and all mods or plugins I have used or provided assistance on the right, but written in Chinese
> 
> I was talking about the Stand options that you mentioned earlier, the game and memory pools. Can you try enabling just one of those and see if you can go online? Can you also try to load a solo session instead of a public session?
> 
> > I use all of these at the same time, about 100 mods, and the original file of mods alone is as high as 23g, not after installation
> > For me, I haven't encountered any problems so far yet
> 
> Well I guess that explains it

I think you reply to the wrong person

**Comment by maybegreat48** [comment]
Can you build [this](https://github.com/maybegreat48/YimASI) project and place the DLL file (`WTSAPI32.dll`) inside your GTA directory? You also have to remove the other ASI loader and use the vanilla ScriptHookV instead (you should be able to go online with it)

**Comment by Tessio** [comment]
> Can you build [this](https://github.com/maybegreat48/YimASI) project and place the DLL file (`WTSAPI32.dll`) inside your GTA directory? You also have to remove the other ASI loader and use the vanilla ScriptHookV instead (you should be able to go online with it)

The problem with yim is because of the .asi, heapadjuster.asi with yimmenu, cause crash to enter in online session, I noticed this after several tests, stand lets me enter online sessions with a custom gameconfig and without modifying the memorypool with the menu, conclusion is because of the compatibility of yimmenu with some .asi

**Comment by Tessio** [comment]
Unfortunately without heapadjuster.asi, the game loads online with yimmenu but has a sync failure warning with the rockstar online catalog and returns you to story mode.

**Comment by Tessio** [comment]
The strange thing is that I can use heapadjuster.asi alone but it does not crashes online, this is quite rare.

**Comment by Tessio** [comment]
> > Can anyone here test #2443?
> 
> It works, i can join a public session with mods, the main issue now is actually loading them because if you load [ClosedIV](https://github.com/martonp96/ClosedIV/) and YimMenu at the same time the game will crash.
> 
> To get it to load, you need to inject ClosedIV with dll hijacking(ASI Loader) and then inject YimMenu with an injector manually. Maybe we can add rpf loading in a later PR.

Something similar to what is mentioned here

**Comment by MsLGXC** [comment]
> > Can you build [this](https://github.com/maybegreat48/YimASI) project and place the DLL file () inside your GTA directory? You also have to remove the other ASI loader and use the vanilla ScriptHookV instead (you should be able to go online with it)`WTSAPI32.dll`
> 
> The problem with yim is because of the .asi, heapadjuster.asi with yimmenu, cause crash to enter in online session, I noticed this after several tests, stand lets me enter online sessions with a custom gameconfig and without modifying the memorypool with the menu, conclusion is because of the compatibility of yimmenu with some .asi

I don't know what you did, but at least it work  well for me

**Comment by MsLGXC** [comment]
i dont think heapadjuster.asi incompatible with yim or i will crash like u

**Comment by MsLGXC** [comment]
![5K06CBD1CAQ` SSTG{ 6~`1](https://github.com/YimMenu/YimMenu/assets/94236627/3e1dbad0-c4f4-4675-8517-f9ebf60864df)
![RVYKJESK0J%GZXM7{IYK(NA](https://github.com/YimMenu/YimMenu/assets/94236627/8205272c-85b4-477d-bd61-79512ae5d78d)
![DEL62W`12V29)X`U{16J FA](https://github.com/YimMenu/YimMenu/assets/94236627/6c3cda61-0568-423c-8a1a-6b8b5cb22cf2)
![LNQWI43NHFP(7WP)(@3T6SI](https://github.com/YimMenu/YimMenu/assets/94236627/eb76de7c-dc4c-41f4-bed1-41b50790e0d8)
![Z9HE7IV ZQ~3ZCV(3AU P~K](https://github.com/YimMenu/YimMenu/assets/94236627/45a09262-e166-4eb6-9cb3-4b036ae84894)
![@EA%{_![5K06CBD1CAQ` SSTG{ 6~`1](https://github.com/YimMenu/YimMenu/assets/94236627/3e1dbad0-c4f4-4675-8517-f9ebf60864df)
![RVYKJESK0J%GZXM7{IYK(NA](https://github.com/YimMenu/YimMenu/assets/94236627/8205272c-85b4-477d-bd61-79512ae5d78d)
![DEL62W`12V29)X`U{16J FA](https://github.com/YimMenu/YimMenu/assets/94236627/6c3cda61-0568-423c-8a1a-6b8b5cb22cf2)
![LNQWI43NHFP(7WP)(@3T6SI](https://github.com/YimMenu/YimMenu/assets/94236627/eb76de7c-dc4c-41f4-bed1-41b50790e0d8)
![Z9HE7IV ZQ~3ZCV(3AU P~K](https://github.com/YimMenu/YimMenu/assets/94236627/45a09262-e166-4eb6-9cb3-4b036ae84894)
QS0{}W$%3Y(4QTN](https://github.com/YimMenu/YimMenu/assets/94236627/a774159f-f0ed-41a8-8996-c2d97424763d)
I'm even playing Liberty City and Vice City

**Comment by MsLGXC** [comment]
For me, not only is there no problem, but his effect has even surpassed that of stand. Because stand did not disable verification for additional dlc, while yim now disabled both dlc check and hash check. This almost allows you to directly install any offline mod and play it directly. Unless certain parameters of the mod itself can cause problems online, 99% of mods can be installed directly through normal methods without the need to modify them to run online like I did before

**Comment by Tessio** [comment]
> For me, not only is there no problem, but his effect has even surpassed that of stand. Because stand did not disable verification for additional dlc, while yim now disabled both dlc check and hash check. This almost allows you to directly install any offline mod and play it directly. Unless certain parameters of the mod itself can cause problems online, 99% of mods can be installed directly through normal methods without the need to modify them to run online like I did before

Well I'm just testing natural vision, so let's wait for more people who want to support the tests and not just enjoy the function already done.

**Comment by Tessio** [comment]
What it does have to do with is something that stand avoids and yimmenu does not

**Comment by maybegreat48** [comment]
Can you elaborate further? What's working for you and what isn't? Are you sure that you're loading WTSAPI32 properly? You should have a YIMAsi log file in the YimMenu directory if it loaded right

**Comment by ewenjam** [comment]
Hi, has there been any recent updates about implementing this? 

I am also trying to get the game to load addon DLC packs (just texture improvements), as well as trying to add simple graphics mods like VisualV. Editing meta files like visualsettings.dat normally makes the game unable to connect with other players like MsLGXC had mentioned in the OP.

Thanks!

**Comment by MsLGXC** [comment]
> Hi, has there been any recent updates about implementing this?
> 
> I am also trying to get the game to load addon DLC packs (just texture improvements), as well as trying to add simple graphics mods like VisualV. Editing meta files like visualsettings.dat normally makes the game unable to connect with other players like MsLGXC had mentioned in the OP.
> 
> Thanks!

It seems that they have abandoned the update. Perhaps it is because the game needs to modify the code every time it is updated? I'm not a professional, so I don't know. It used to work well, but currently cannot inject old DLLs in the latest version. I am still looking for a universal way for everyone to load all mods without using any menus. It is still unrealistic to require everyone to use yim to load. Some people are always afraid of the security of the menu and not everyone has the patience to perform a complex operation to play with

**Comment by MsLGXC** [comment]
> Hi, has there been any recent updates about implementing this?
> 
> I am also trying to get the game to load addon DLC packs (just texture improvements), as well as trying to add simple graphics mods like VisualV. Editing meta files like visualsettings.dat normally makes the game unable to connect with other players like MsLGXC had mentioned in the OP.
> 
> Thanks!

You can use vsreloader to hot load vssettings during gameplay. Hot loading will not cause asset hash check. It seems that it only calculates when the game starts maybe.u need rebuild vsreloader or that will be crash because havent update.2t1hook can do this then delete 2t1hook and everything related to it to make the native shv work properly, otherwise the game will crash like before.You can also use ready-made versions, such as nveo from zzoko or in my mod too

**Comment by ewenjam** [comment]
> > Hi, has there been any recent updates about implementing this?
> > I am also trying to get the game to load addon DLC packs (just texture improvements), as well as trying to add simple graphics mods like VisualV. Editing meta files like visualsettings.dat normally makes the game unable to connect with other players like MsLGXC had mentioned in the OP.
> > Thanks!
> 
> You can use vsreloader to hot load vssettings during gameplay. Hot loading will not cause asset hash check. It seems that it only calculates when the game starts maybe

Ah ok, interesting that vsreloader can bypass it, thank you!

**Comment by ewenjam** [comment]
> > Hi, has there been any recent updates about implementing this?
> > I am also trying to get the game to load addon DLC packs (just texture improvements), as well as trying to add simple graphics mods like VisualV. Editing meta files like visualsettings.dat normally makes the game unable to connect with other players like MsLGXC had mentioned in the OP.
> > Thanks!
> 
> You can use vsreloader to hot load vssettings during gameplay. Hot loading will not cause asset hash check. It seems that it only calculates when the game starts maybe

I tried using VSReloader (an updated version, not the one from 5-Mods), but it refuses to load through dsound.dll. Im guessing its either not updated for GTA 1.68 or Im missing some step. Have you tried it recently? Ty!

**Comment by Yimura** [comment]
Implemented in #3274
